### PR TITLE
Fix 404.html with locales

### LINKED ISSUE #67: BUG: Spanish word in Portuguese locale for site footer
# Bug Report
## Describe the bug

In Spanish word "impulsado" has been introduced into the Portuguese locale file in #51.

https://github.com/pacollins/hugo-future-imperfect-slim/blob/80ca0db9e4e71560508381b001b18bde5313d4a1/i18n/br.toml#L50

However, this is not a Portuguese word.  You may see, for example, [Liguee](https://www.linguee.es/espanol-portugues/search?source=portugues&query=impulsado), [Wiktionary](https://pt.wiktionary.org/wiki/Especial:Pesquisar?search=Impulsado&go=Definir&ns0=1), [Collins Dictionary](https://www.collinsdictionary.com/spellcheck/portuguese-english?q=impulsado), etc.

## Expected behavior

A Portuguese translation for "powered by".

## Screenshots

NIL

## Environment

 - **Hugo Version:** [e.g. 0.46]
 - **Browser and Version:** [e.g. Chrome 74.0.3729.108]
 - **Platform:** [e.g. Desktop/Mobile]
 - **Link to Project:**

### LINKED ISSUE #70: BUG: Missing locales for 404 page
# Bug Report
## Describe the bug

https://github.com/pacollins/hugo-future-imperfect-slim/blob/58c2de987481f7bb3d3d37500186c69db633efe8/layouts/404.html#L56-L59

Since our theme claims to be a "multilingual blogging theme for Hugo", we've to fix that.

## Expected behavior

Translations for these strings.

### PR DESCRIPTION
**Pull request by pacollins**
## Description

i18n added to `404.html`.

## Motivation and Context

Fixes #70 and fixes #67 

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have read the [Contributing Document](https://github.com/pacollins/hugo-future-imperfect-slim/blob/master/.github/CONTRIBUTING.md).
- [X] My code follows the [code style](https://github.com/pacollins/hugo-future-imperfect-slim/blob/master/.github/CONTRIBUTING.md#Style-Guide) of this project.
- [ ] My change requires a change to the [documentation](https://github.com/pacollins/hugo-future-imperfect-slim/wiki).
- [ ] I have updated the documentation, including `theme.toml`, accordingly.

### CODE REVIEW

**Comment by VincentTam** [comment]
Faster than I do.

```diff
diff --git a/i18n/br.toml b/i18n/br.toml
index 940e46d..1f75bb7 100644
--- a/i18n/br.toml
+++ b/i18n/br.toml
@@ -30,6 +30,8 @@
   other = "horas"
 [submit]
   other = "Enviar"
+[404_page_not_found]
+  other = "Página Não Encontrada"
 [404_oops]
   other = "Oops, a páginas que você está<br/>procurando não existe."
 [404_backorreport]
diff --git a/i18n/en.toml b/i18n/en.toml
index b9f0d7d..7d07467 100644
--- a/i18n/en.toml
+++ b/i18n/en.toml
@@ -30,6 +30,8 @@
   other = "hours"
 [submit]
   other = "Submit"
+[404_page_not_found]
+  other = "Page Not Found"
 [404_oops]
   other = "Oops, the page you're<br/>looking for does not exist."
 [404_backorreport]
diff --git a/i18n/es.toml b/i18n/es.toml
index f069c47..41fa9ca 100644
--- a/i18n/es.toml
+++ b/i18n/es.toml
@@ -30,6 +30,8 @@
   other = "horas"
 [submit]
   other = "Enviar"
+[404_page_not_found]
+  other = "Página No Encontrada"
 [404_oops]
   other = "Ups, lapágina que estas <br/>buscando no exite."
 [404_backorreport]
diff --git a/i18n/fr.toml b/i18n/fr.toml
index 8425992..b111cd5 100644
--- a/i18n/fr.toml
+++ b/i18n/fr.toml
@@ -30,6 +30,8 @@
   other = "heures"
 [submit]
   other = "Envoyer"
+[404_page_not_found]
+  other = "Page Non Trouvée"
 [404_oops]
   other = "Oops, la page que vous<br/>demandez n'existe pas."
 [404_backorreport]
diff --git a/i18n/pl.toml b/i18n/pl.toml
index d7ef6e1..334c834 100644
--- a/i18n/pl.toml
+++ b/i18n/pl.toml
@@ -30,6 +30,8 @@
   other = "Godziny"
 [submit]
   other = "Prześlij"
+[404_page_not_found]
+  other = "Strona Nie Znaleziona"
 [404_oops]
    other = "Ups, strona, którą szukasz <br/>, nie istnieje."
 [404_backorreport]
diff --git a/i18n/zh-CN.toml b/i18n/zh-CN.toml
index c8e5732..8eb58cc 100644
--- a/i18n/zh-CN.toml
+++ b/i18n/zh-CN.toml
@@ -30,6 +30,8 @@
   other = "小时"
 [submit]
   other = "送出"
+[404_page_not_found]
+  other = "网页未找到"
 [404_oops]
   other = "噢，您索要的页面<br/>不存在。"
 [404_backorreport]
diff --git a/i18n/zh-TW.toml b/i18n/zh-TW.toml
index 91dbe6d..ccee5d0 100644
--- a/i18n/zh-TW.toml
+++ b/i18n/zh-TW.toml
@@ -30,6 +30,8 @@
   other = "小時"
 [submit]
   other = "送出"
+[404_page_not_found]
+  other = "網頁未找到"
 [404_oops]
   other = "噢，您索要的頁面<br/>不存在。"
 [404_backorreport]
diff --git a/layouts/404.html b/layouts/404.html
index 1728411..fbca45a 100644
--- a/layouts/404.html
+++ b/layouts/404.html
@@ -1,62 +1,18 @@
 <!doctype html>
 <html lang="en">
-<head>
+  <head>
     <meta charset="utf-8">
-    <title>Page Not Found</title>
+    <title>{{ i18n "404_page_not_found" }}</title>
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <style>
-
-        * {
-            line-height: 1.2;
-            margin: 0;
-        }
-
-        html {
-            color: #888;
-            display: table;
-            font-family: sans-serif;
-            height: 100%;
-            text-align: center;
-            width: 100%;
-        }
-
-        body {
-            display: table-cell;
-            vertical-align: middle;
-            margin: 2em auto;
-        }
-
-        h1 {
-            color: #555;
-            font-size: 2em;
-            font-weight: 400;
-        }
-
-        p {
-            margin: 0 auto;
-            width: 280px;
-        }
-
-        @media only screen and (max-width: 280px) {
-
-            body, p {
-                width: 95%;
-            }
-
-            h1 {
-                font-size: 1.5em;
-                margin: 0 0 0.3em;
-            }
-
-        }
-
-    </style>
-</head>
-<body>
+    <link rel='stylesheet' type='text/css' href='{{ "css/404.css" | relURL }}'>
+  </head>
+  <body>
     <h1>404</h1>
-    <h2>Page Not Found</h2>
-    <p>Sorry, but the page you were trying to view does not exist.</p>
-    <a href="javascript:history.back()">Go Back</a>
-</body>
+    <h2>{{ i18n "404_page_not_found" }}</h2>
+    <p>{{ i18n "404_oops" | safeHTML }}</p>
+    <p>{{ i18n "404_backorreport" | safeHTML }}</p>
+    <a href='javascript:history.back()'>{{ i18n "404_gotohome" }}</a>&nbsp;&nbsp;
+    <a href='{{ "contact" | relLangURL }}'>{{ i18n "404_report" }}</a>
+  </body>
 </html>
 <!-- IE needs 512+ bytes: https://blogs.msdn.microsoft.com/ieinternals/2010/08/18/friendly-http-error-pages/ -->
```

File: i18n/br.toml [hunk 1]
```diff
@@ -30,14 +30,12 @@
...
-[404_backorreport]
-  other = "Você pode querer voltar à homepage.<br/>Se você acha que este é um erro, informe o problema."
-[404_gotohome]
-  other = "Ir para Homepage"
-[404_report]
-  other = "Informar um problema"
+[404_pagenotfound]
+  other = "Página Não Encontrada"
+[404_sorry]
+  other = "Desculpe, mas a página que você estava tentando visualizar não existe."
```
Comment (VincentTam) [inline i18n/br.toml]: I don't know Portuguese, but copying from existing translation like https://codingmall.com/pt/page-not-found is more efficient.

File: i18n/es.toml [hunk 2]
```diff
@@ -30,14 +30,12 @@
...
-[404_backorreport]
-  other = "Seguramente quieras vover a la página principal.<br/>Si crees que algo no va bien, avísalo."
-[404_gotohome]
-  other = "Ir a la página principal"
-[404_report]
-  other = "Avisar de un problema"
+[404_pagenotfound]
+  other = "Página No Encontrada"
+[404_sorry]
+  other = "Lo sentimos, pero la página que intentaba ver no existe."
```
[the remark this answers is not in the document]
Comment (reymon359) [reply i18n/es.toml]: The translation is pretty good that way, maybe I would change the sorry part with ```Lo sentimos, la página que estas buscando no existe``` to be more accurate.

File: layouts/404.html [hunk 3]
```diff
@@ -54,9 +54,9 @@
 </head>
 <body>
     <h1>404</h1>
-    <h2>Page Not Found</h2>
-    <p>Sorry, but the page you were trying to view does not exist.</p>
-    <a href="javascript:history.back()">Go Back</a>
+    <h2>{{ i18n "404_pagenotfound" . | safeHTML }}</h2>
+    <p>{{ i18n "404_sorry" . | safeHTML }}</p>
+    <a href="javascript:history.back()">{{ i18n "404_goback" . | safeHTML }}</a>
```
[the remark this answers is not in the document]
Comment (VincentTam) [reply layouts/404.html]: Now, I understand.  The few lines above use `safeHTML`.  Let's drop this question.

**Comment by pacollins** [comment]
@VincentTam and @reymon359 I will gladly make the edits, but you should be able to check-out this PR/Branch and push commits to it (that way you will get credit for your contributions 😄). I will work on this later today.

**Comment by reymon359** [comment]
Done!👌

**Comment by VincentTam** [comment]
I'm going to merge this in a hour if everything is good.

**Pull request merged**